script.js

```diff
@@ -0,0 +1,333 @@
+    }
+
+    if (item) {
+      this._spatialNavigator.focus(item);
+      return true;
+    }
+
+    return false;
+  };
+
+  /**
+   * Move the focus element
+   */
+  proto.move = function(direction) {
+    return this._spatialNavigator.move(direction);
+  };
+
+  /**
+   * Add navigable elements in scrollbox
+   */
+  proto.addNavigableElems = function(navigableElems) {
+    return this._spatialNavigator.multiAdd(navigableElems);
+  };
+
+  /**
+   * Remove navigable elements in scrollbox
+   */
+  proto.removeNavigableElems = function(navigableElems) {
+    return this._spatialNavigator.multiRemove(navigableElems);
+  };
+
+  proto.handleEvent = function (evt) {
+
+    switch (evt.type) {
+      case 'transitionend':
+        if (evt.target === this && evt.propertyName === 'transform') {
+          this._fireEvent('scrollbox-scroll', {
+            element: this
+          });
+        }
+        break;
+      case 'scroll':
+        // Gecko may scroll the scrollbox automatically, we want to prevent this
+        // behavior inorder to have correct view.
+        if (evt.target === this) {
+          evt.target.scrollLeft = 0;
+        }
+        break;
+    }
+  };
+
+  /**
+   * Get the node element of the item element
+   */
+  proto._getNodeFromItem = function(itemElem) {
+    if (!itemElem) {
+      return null;
+    }
+
+    var nodeElem = itemElem;
+    // Find the node element. If the node is removed from scrollbox, it's parent
+    // will be null, so we may handle the case here.
+    while (nodeElem.parentElement && nodeElem.parentElement !== this) {
+      nodeElem = nodeElem.parentElement;
+    }
+
+    return nodeElem;
+  };
+
+  /**
+   * Scroll the list to the item element
+   */
+  proto._scrollTo = function(itemElem) {
+    this.translateX = this._getScrollOffset(this._getNodeFromItem(itemElem));
+    this.listElem.style.transform = 'translateX(' + this.translateX + 'px)';
+  };
+
+
+  /**
+   * Get the offset of the node element
+   */
+  proto._getScrollOffset = function(nodeElem) {
+    var nodeLeft = nodeElem.offsetLeft;
+    var nodeWidth = nodeElem.offsetWidth;
+    var listWidth = this.listElem.offsetWidth;
+    var newTranslate = this.translateX;
+
+    if (listWidth < this.viewWidth) {
+      // scroll to center
+      return (this.viewWidth - listWidth) / 2 + this.margin;
+    } else if (nodeLeft + nodeWidth >
+                          -this.translateX + this.viewWidth + this.margin) {
+      // scroll left
+      newTranslate = this.viewWidth - nodeLeft - nodeWidth + this.margin;
+    } else if (nodeLeft < -this.translateX + this.margin) {
+      // scroll right
+      newTranslate = -nodeLeft + this.margin;
+    }
+
+    // If the new scroll offset contains first/last node, we have to align the
+    // list to begin/end.
+    if (this.lastElementChild.offsetLeft + this.lastElementChild.offsetWidth <=
+                             -newTranslate + this.viewWidth + this.margin) {
```

Comment:
I'm thinking if we can use `if (nodeElem == this.firstElementChild)` and `if (nodeElem === this.firstElementChild)` here